text/0000-PXE-Booting-UEFI.md

```diff
@@ -0,0 +1,228 @@
+is an open item).
+
+Orchestration TFTP code must be modified to use multiple "KIND LOADER default
+local boot" and "KIND LOADER global default" templates. Local and global boot
+templates must be added for Grub2 new kind. Naming convention will be:
+
+* PXELinux BIOS default local boot
+* PXELinux UEFI default local boot
+* PXEGrub BIOS default local boot
+* PXEGrub UEFI default local boot
+* etc. (the same for global default templates)
+
+But only kind/variant is not enough, users need to select BIOS or UEFI booting
+method. This is where Loader flag comes in - selection of three constants:
+BIOS, UEFI and SecureBoot UEFI. The first two are obvious causing selection of
+the correct DHCP filename when deploying TFTP configuration, the third one
+select "shim.efi" chain loading capability for secure booting.
+
+Similarly to root password, a global setting could be provided for one or both
+flags for missing values (open item).
+
+### [Foreman installer #14920](http://projects.theforeman.org/issues/14920) - Change dhcpd.conf to support EFI loaders
+
+Default DHCPD configuration must be changed in order to allow booting unknown
+UEFI hosts as it was shown above (dhcpd.conf snippet). Using shim.efi is
+perhaps better default than loading Grub directly.
+
+### [Foreman installer #12635](http://projects.theforeman.org/issues/12635) - Options to deploy Grub and PXELinux EFI loaders in TFTP root
+
+It looks like Grub2 is the best and most compatible way of PXE booting EFI
+hardware and it is available in most Linux distributions. It makes sense to
+provide support in our Puppet classes and installer to deploy these.
+Instructions on where to find the firmware in RHEL7 are at
+[access.redhat.com](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/Installation_Guide/chap-installation-server-setup.html)
+(RHEL6 does not ship Grub2 so installer should fail or skip that). Similarly
+for Debian and Ubuntu it is available for [direct
+download](http://archive.ubuntu.com/ubuntu/dists/trusty/main/uefi/grub2-amd64/current/)
+or in debian-installer-8-netboot-amd64 package.
+
+UEFI shim shall be installed if possible. Expected file paths are defined in
+the table above.
+
+### [Foreman core #14921](http://projects.theforeman.org/issues/14921) - ComputeResource should pass host loader flag to VM
+
+Since host will have "loader" (BIOS/EFI/SecureBoot EFI) flag, compute resources
+could pass this information when creating new VMs so they gets created with
+correct firmware and PXE booting will just work.
+
+### [Foreman discovery #14922](http://projects.theforeman.org/issues/14922) - Discovery can detect BIOS/EFI and report it back
+
+Discovery can detect current OS loader via a [custom
+fact](https://github.com/jcpunk/puppet-efi/blob/master/lib/facter/efi.rb) and
+this can be used for provisioning to set the "loader" host attribute. Detection
+of secure boot should be possible too, it will be more challenging (depends on
+distribution).
```

Comment:
Yeah, scratch that. I only need to investigate how to determine secure boot on Fedora/CentOS.